src/acvp_kas_ecc.c

```diff
@@ -125,13 +156,29 @@ static ACVP_RESULT acvp_kas_ecc_init_cdh_tc (ACVP_CTX *ctx,
                                              char *psy,
                                              unsigned int mode
 ) {
-
+    ACVP_RESULT rv;
     stc->mode = mode;
 
     stc->psx = calloc(1, ACVP_KAS_ECC_MAX_STR);
     if (!stc->psx) { return ACVP_MALLOC_FAIL; }
+    rv = acvp_hexstr_to_bin((const unsigned char *) psx, stc->psx, ACVP_KAS_ECC_MAX_STR, &(stc->psxlen));
```

Comment:
This could probably use a #define MAX_BYTE to cut the allocated memory in half. General comment